V20 from swang373

perrozzi opened the pull request.

Commit:
added > 10 GeV in ttbar CR to protect from low-mass resonances

Commit:
Saved configuration for Hbb meeting 30/03/16

Commit:
Set-up config to run on preptest2. Added modifications suggested by Luca

Commit:
modifed ratio to 0.5-1.75. Should now be sync with David

Commit:
.

Commit:
Added macro to evaluate size of PREPout samples for difference cuts

silviodonato committed:
Zvv stuff

silviodonato committed:
Zvv stuff

Commit:
Included lumi for mu trigg calculation

silviodonato committed:
adding some empty folders

Commit:
Made order in cut config

Commit:
switch to reg. variables in the CR/SR

Commit:
added ggZH. updated signal cross-sections

swang373 committed:
Organize and manually parse configuration files

This has been personally instructive to see what options exist and
the logical flow of the configuration. I have added a modified steps
folder, changing how the commands are called because that many jobs
cannot be launched in the background on lxplus machines.

swang373 committed:
New das_query.py script

This script queries DAS to find the LFNs for the files of those samples
defined in samples_nosplit.ini.

The positional argument "sample_config" is the samples_nosplit.ini file
storing the query options and primary dataset names. The optional
argument "--outdir" is a name for the subdirectory created within the
configuration directory to hold the lists of LFNs. If not passed, the
LFN lists will be saved directly within the configuration directory.

Rather than importing the das_client as a module, the script calls the
das_client through a subprocess. This design choice was made because the
main() function in das_client.py does not take an argv argument, which
means taking advantage of its convenient features requires tampering
with sys.argv. Since the das_client is called from within a script which
parses sys.argv for command line arguments, calling a subprocess with a
set list of arguments avoids setting and restoring sys.argv manually.

swang373 committed:
Modify das_query options, start writing prep.py

The --outdir option of das_query has been removed. Not only will this
option be used all the time, but a consistent place to specify it is
needed to propagate it to other scripts as needed. It has since been
promoted to a configuration file option in "samples_nosplit.ini"

A first pass at a rework of treecopier has been started, with an
accompanying prep.py script for running the preparation step in the
context of lxplus.

Commit:
Saved configuration for Hbb metting 07/04/2016

Commit:
add all stats from signal MC

perrozzi commented:
@swang373 may I merge?

swang373 commented:
@perrozzi

Unfortunately, not yet. I'm making some updates to that branch still. I'll
let you know when it's ready sometime this evening! Thanks for the reminder!

On Thu, Apr 7, 2016 at 2:50 PM, perrozzi notifications@github.com wrote:

> @swang373 https://github.com/swang373 may I merge?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> https://github.com/perrozzi/Xbb/pull/59#issuecomment-206877246

swang373 committed:
Reworked TreeCopier Class, prep.py for lxplus

The TreeCopier class was reworked to parse from a list of LFNs, removing
the awkward subprocess call to gfalFS. However, testing the TreeCopier
class revealed fragility in its use of multiprocessing.

- Threads and multiple processes should not be mixed in Python, and the
  logging module uses a thread which is known to cause multiple
  processes to lock up. The logging module has been dropped, and logging
  statements simply use print.
- The multiple processes no longer target a class method of TreeCopier.
  They are now subclassed from multiprocessing.Process. This was done to
  ensure that shared state was avoided. The multiprocessing module has
  its own fork of pickle which can serialize class methods, and I wasn't
  modifying class attributes, but I've factored this out to be safe.

swang373 committed:
Add more content to config files, minor tweaks

Besides more comments to document the configuration files, the
xrd_redirector and merge arguments to the TreeCopier class have been
promoted to configuration options in samples_nosplit.ini.

Also, rather than having a file named "nothing" inside the LOG/
directory, I thought it'd be more useful as a .gitignore that tells git
to ignore any files in the directory which end with ".log".

swang373 committed:
Add README to document my progress

perrozzi commented:
ps there is something strange with this PR, it contains commits from Gael and even myself (?!?!) which I should already have. closing and creating another one to check what happens
